astacus/coordinator/plugins/clickhouse/object_storage.py

```diff
@@ -159,6 +177,14 @@ def delete_item(self, key: str) -> None:
 
     def copy_items_from(self, source: "ObjectStorage", keys: Sequence[str], *, stats: StatsClient | None) -> None:
         keys_set = set(keys)
-        for source_item in source.list_items():
+        keys_to_copy = len(keys_set)
+        for keys_copied, source_item in enumerate(source.list_items(), start=1):
             if source_item.key in keys_set:
                 self.items[source_item.key] = source_item
+            if stats:
+                _emit_copy_progress_metric(
```

Comment:
maybe just move the `MemoryObjectStorage` into `tests`? since it's not used in prod, emitting metrics from it does not add much value; but since it's near the prod code, one is tempted to add prod behavior to it

Reply:
Moved to `tests/unit/coordinator/plugins/clickhouse/object_storage.py` in a separate commit.